src/Disks/ObjectStorages/S3/registerDiskS3.cpp

```diff
@@ -65,7 +65,34 @@ void checkReadAccess(const String & disk_name, IDisk & disk)
         throw Exception("No read access to S3 bucket in disk " + disk_name, ErrorCodes::PATH_ACCESS_DENIED);
 }
 
-void checkRemoveAccess(IDisk & disk) { disk.removeFile("test_acl"); }
+void checkRemoveAccess(IDisk & disk)
+{
+    disk.removeFile("test_acl");
+}
+
+bool checkBatchRemoveIsMissing(S3ObjectStorage & storage, const String & key)
+{
+    String path = key + "/test_acl";
+    auto file = storage.writeObject(path, WriteMode::Rewrite);
+    try
+    {
+        file->write("test", 4);
+    }
+    catch (...)
+    {
+        file->finalize();
+        return false; /// We don't have write access, therefore no information about batch remove.
+    }
+    try
+    {
+        storage.removeObjects({{ path, 0 }});
```

Comment:
Worth to add a comment that it uses `DeleteObjects` not `DeleteObject`.
Also I think it worth document `removeObjects` in header, add a note about which API it uses, about `support_batch_delete` and GCS.